src/nbt/compound.rs

```diff
@@ -3,23 +3,23 @@ use bytes::{Buf, BufMut, Bytes, BytesMut};
 use crab_nbt::nbt::tag::NbtTag;
 use crab_nbt::nbt::utils::{get_nbt_string, END_ID};
 use derive_more::Into;
-use std::collections::{hash_map::IntoIter, HashMap};
 use std::io::{Cursor, Write};
+use std::vec::IntoIter;
 
 #[derive(Clone, PartialEq, Debug, Default, Into)]
 pub struct NbtCompound {
-    pub child_tags: HashMap<String, NbtTag>,
+    pub child_tags: Vec<(String, NbtTag)>,
 }
 
 impl NbtCompound {
     pub fn new() -> NbtCompound {
         NbtCompound {
-            child_tags: HashMap::new(),
+            child_tags: Vec::new(),
```

Comment:
```suggestion
            child_tags: vec![],
```

Reply:
Nope. We used a lot of vec new in our codebase